src/org/exist/indexing/IndexController.java

```diff
@@ -300,12 +299,12 @@ public StreamListener getStreamListener(final DocumentImpl doc, final ReindexMod
      */
     public StreamListener getStreamListener() {
         if (listener != null) {
-            StreamListener next = listener;
-            while (next != null) {
-                // wolf: setDocument() should have been called before
-                // next.getWorker().setDocument(currentDoc, currentMode);
-                next = next.getNextInChain();
-            }
+//            StreamListener next = listener;
```

Comment:
for what? that is local variable. That block did `next.getWorker().setDocument`, without that it do nothing.